spec/factories/vm_or_template.rb

```diff
@@ -1,5 +1,5 @@
 FactoryBot.define do
-  factory :vm_or_template do
+  factory :vm_or_template, :class => "ManageIQ::Providers::Vmware::InfraManager::Vm" do
```

Comment:
we have a separate factory vm_vmware which already does this.

Reply:
I was thinking that vm, vm_or_template, and vm_vmware would be interchangeable. They instantiate a vm.

Or maybe they pick a random child? (though we need consistency with that and the corresponding ems)

I do like when tests link to "any vm or template".
But I don't think we should instantiate that class.

I couldn't declare the parent of vm to be vm_vmware since we needed it to still flow up to vm_or_template.